aten/src/ATen/core/function_schema.h

```diff
@@ -639,6 +639,40 @@ template<>
       return c10::hash_combine(std::hash<size_t>()(arg.index), std::hash<size_t>()(static_cast<std::size_t>(arg.type)));
     }
   };
+template<>
+  struct hash<c10::Argument> {
+    size_t operator()(const c10::Argument& arg) const
+    {
+      // We don't need to hash default value and alias_info
+      // because they are not useful in distinguishing schema.
```

Comment:
The hash algorithm should respect equality?

Reply:
Like, if you have FunctionSchemas with differing default values hash to the same thing, then why did you just hash on the name of the schema only. Would be a lot simpler!

Reply:
I see, yeah so far I don't think we have two schemas with exact same arguments and only differs in default values, and couldn't reason about having such ones in the future, that was the primary reason I chose to not hash default value. But if we want to be more respectful to equality, I can surely add the default value hash. I think we still want to leave out the alias_info as it contains a ton of information which is difficult to hash and not really useful in distinguishing schemas. 

I'll update the PR to include default value hashing.